app/controllers/concerns/search_aggregator.rb

```diff
@@ -228,5 +243,26 @@ def blacklist_cls_id_components(cls_id, blacklist_words)
 
     stripped_id
   end
-end
 
+  def merge_federated_results(search_results)
+    search_results.each do |element|
+      element[:root][:other_portals] = []
+      element[:reuses].reject! do |reuse|
+        if element[:root][:title] == reuse[:root][:title]
```

Comment:
the merging need to be done by concept IDs and ontology acronym, not  by prefLabel and ontology acronym.